lib/index.js

```diff
@@ -149,7 +151,7 @@ function getOptions(options) {
 
   options.error = function(err) {
     if (error) {
-      error(util._extend(new Error(), JSON.parse(err)));
+      error.call(options.context, util._extend(new Error(), JSON.parse(err)));
```

Comment:
Since we are not cloning `options` on `context` instantiation, I think one problem with this approach would be the shallow copy dropping functions definitions (`importer` et al.).